apps/explorer/lib/explorer/celo/epoch_util.ex

```diff
@@ -12,11 +12,18 @@ defmodule Explorer.Celo.EpochUtil do
     rem(bn, 17_280) == 0
   end
 
-  def calculate_epoch_transaction_count_for_block(bn, nil = _epoch_rewards), do: 0
+  def calculate_epoch_transaction_count_for_block(_, nil), do: 0
 
-  def calculate_epoch_transaction_count_for_block(bn, _epoch_rewards) do
+  def calculate_epoch_transaction_count_for_block(bn, epoch_rewards) do
     if is_epoch_block?(bn) do
-      CeloElectionRewards.get_epoch_transaction_count_for_block(bn) + 2
+      additional_transactions_count =
+        if Decimal.cmp(epoch_rewards.reserve_bolster.value, 0) == :gt do
+          3
+        else
+          2
+        end
+
+      CeloElectionRewards.get_epoch_transaction_count_for_block(bn) + additional_transactions_count
```

Comment:
why are we adding to the number of transactions here?

Reply:
Because reserve bolster / community fund are not real `celo_election_rewards` that are stored in the database, rather they're derived from the data in `celo_epoch_rewards`. We're pushing them on top of the list.

Reply:
I don't understand the concepts being modelled here. Epoch transactions are essentially events emitted by the blockchain client directly as part of the celo protocol - essentially logs where the transaction hash is equal to the block hash - of which the "bolster" `Transfer` event and   

For what purpose are we adding to this count?

Reply:
It was there in the original implementation, we can think about refactoring it separately, but IMHO way out of scope of that ticket.

Reply:
Ok, but again, for what purpose are we adding to this count? Why do we need to increase this value artificially?

Reply:
Because we have 2 separate sources of data for epoch transactions. One is in `celo_election_rewards` where we have multiple rows per epoch block (for voters, validators etc) and one is in `celo_epoch_rewards` where we have a single row for any epoch block. And from that one row we're kinda "extracting" multiple transactions to show them in the form of transaction tiles.

Reply:
For reference: https://github.com/celo-org/blockscout/pull/670/files#r905146073

Reply:
ok so this value is used in the UI for the number of tiles to show, and therefore needs to represent the size of the list in order to display all the data (?) 

That doesn't sound appropriate to me, but given the nature of the PR you are right that refactoring is out of scope. I wanted to know why the behavior had to change, thanks for explaining.

Reply:
I agree 100% with you, created a follow up ticket for it: https://github.com/celo-org/data-services/issues/415